neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -26,12 +26,20 @@ public class SetFilter { public BloomFilter Filter; }
         private VersionPayload version;
         private bool verack = false;
         private BloomFilter bloom_filter;
+        private IActorRef parent;
+
 
         public ProtocolHandler(NeoSystem system)
         {
             this.system = system;
             this.knownHashes = new FIFOSet<UInt256>(Blockchain.Singleton.MemPool.Capacity * 2);
             this.sentHashes = new FIFOSet<UInt256>(Blockchain.Singleton.MemPool.Capacity * 2);
+            this.parent = Context.Parent;
+        }
+
+        public ProtocolHandler(NeoSystem system, IActorRef parent) : this(system)
```

Comment:
I don't like this. It seems to have not been used in any place other than testing. You can create `ProtocolHandler` in test, and it will automatically set the `Parent` for actors.

Reply:
Lets try to put it more transparent Ricardo.
Erik the big challenge is that actors are not the same, on real code, and on testing.